btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -539,6 +539,10 @@ func (sew *StakingEventWatcher) checkBtcForStakingTx() error {
 	}
 
 	for _, del := range delegations {
+		if del.ActivationInProgress {
```

Comment:
Would be better if we didn't get all the tracked delegations at once, because it could happen that we load the delegations to iterate, after that someone updates the `del.ActivationInProgress` value and this iterator would not know about it 

If we are okay with this scenario the code looks good to me

Reply:
I do agree with what you say. Tho I do think there's not a high chance for this to happen. What approach would you recommend, first thing that come to my mind was using Iterators

Reply:
yes, like an iterator would work or a function which gets the next del that is not activated

Reply:
will create an issue to improve this, not a high priority, worst case is we retry